src/GraphQL.MicrosoftDI/ValidateServicesSchemaValidator.cs

```diff
@@ -0,0 +1,90 @@
+using System.Reflection;
+using GraphQL.Types;
+using GraphQL.Utilities;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI;
+
+/// <summary>
+/// A schema validator that ensures all services required by <see cref="FromServicesAttribute"/> are registered.
+/// </summary>
+internal sealed class ValidateServicesSchemaValidator : BaseSchemaNodeVisitor
+{
+    private readonly Func<Type, bool> _isValidService;
+    private List<string>? _errors;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ValidateServicesSchemaValidator"/> class.
+    /// </summary>
+    private ValidateServicesSchemaValidator(Func<Type, bool> isValidService)
+    {
+        _isValidService = isValidService;
+    }
+
+    /// <inheritdoc/>
+    public override void VisitObjectFieldDefinition(FieldType field, IObjectGraphType type, ISchema schema)
+    {
+        var serviceTypes = field.GetMetadata<List<Type>>(FromServicesAttribute.REQUIRED_SERVICES_METADATA);
+        if (serviceTypes != null)
+        {
+            foreach (var serviceType in serviceTypes)
+            {
+                if (!_isValidService(serviceType))
+                {
+                    _errors ??= new();
+                    _errors.Add($"The service '{serviceType.FullName}' required by '{type.Name}.{field.Name}' is not registered.");
+                }
+            }
+        }
```

Comment:
I think it's also worth clearing the `REQUIRED_SERVICES_METADATA` to allow the GC. Maybe in `PostVisitSchema`

Reply:
Not so sure about this.  We don't do this kind of cleanup anywhere else in the codebase, rather leaving the metadata static for the lifetime of the schema.  For example, every field that is defined by an expression, such as `Field(x => x.Name)`, stores a reference to the `PropertyInfo` used by by the expression.  It is not used by the schema after the field resolver is compiled.  In addition, the memory consumption of this metadata should be extremely minimal compared to the memory footprint of the schema, and speed isn't a concern as metadata isn't used during execution.  Do you really think this is worthwhile?

Reply:
No, not really. This would be a micro optimization and if we do that here, we need to do that everywhere.

Reply:
We could optimize the memory requirements for this metadata by storing the Type directly for a single entry, and only creating a List when there is a second entry.  Another optimization could be to always store an array, at the cost of recreating the array for each additional member.